src/formatting/expr.rs

```diff
@@ -1985,6 +1948,28 @@ pub(crate) fn rewrite_assign_rhs_with<S: Into<String>, R: Rewrite>(
     Some(lhs + &rhs)
 }
 
+pub(crate) fn rewrite_assign_rhs_with_comments<S: Into<String>, R: Rewrite>(
+    context: &RewriteContext<'_>,
+    lhs: S,
+    ex: &R,
+    shape: Shape,
+    rhs_tactics: RhsTactics,
+    between: Span,
+) -> Option<String> {
+    let lhs = lhs.into();
+    let rhs = rewrite_assign_rhs_expr(context, &lhs, ex, shape, rhs_tactics)?;
+    let comment_shape = shape.block_left(context.config.tab_spaces())?;
+
+    let comment = context.snippet_provider.span_to_snippet(between)?;
+    if comment.trim().starts_with("/") {
+        let rhs = rhs.trim_start();
+
+        combine_strs_with_missing_comments(context, &lhs, &rhs, between, comment_shape, false)
+    } else {
+        Some(lhs + &rhs)
+    }
+}
```

Comment:
Remember that the rhs isn't always going to be a lit, so if we have an inbetween comment then we should probably go ahead and use an updated shape to pass to `rewrite_assign_rhs_expr`

Consider this for example:
```rust
const B: usize =    // baz
            /* ba */
 {  3  
 // foo   
    };
```

We know that the start of the expression is going to end up on the next line due to the inbetween comments with newlines, but the rhs was still formatted with the original shape and (incorrectly) thinks that the opening brace will end up on the same line as the lhs.

What about tweaking what you have a bit to something like this:
```suggestion
    let lhs = lhs.into();
    let contains_comment = contains_comment(context.snippet(between_span));
    let shape = if contains_comment {
        shape.block_left(context.config.tab_spaces())?
    } else {
        shape
    };
    let rhs = rewrite_assign_rhs_expr(context, &lhs, ex, shape, rhs_tactics)?;
    
    if contains_comment {
        combine_strs_with_missing_comments(context, &lhs, &rhs, between, comment_shape, *new_param_or_true*)
    } else {
        Some(lhs + &rhs)
    }
}
```